sdk/storage/Azure.Storage.Common/tests/StorageBearerTokenChallengeAuthorizationPolicyTests.cs

```diff
@@ -100,5 +102,46 @@ public async Task SendsUnUnAuthroizedRequestWhenEnableTenantDiscoveryIsTrue([Val
             // If enableTenantDiscovery is true, all but the first request should be authorized.
             Assert.True(transport.Requests.Skip(enableTenantDiscovery ? 1 : 0).All(r => r.Headers.TryGetValue("Authorization", out _)));
         }
+
+        [Test]
+        public async Task UsesScopeFromBearerChallange()
+        {
+            // Arrange
+            bool firstRequest = true;
+            string initialMismatchedScope = "https://disk.compute.azure.com/.default";
+            string serviceChallengeResponseScope = "https://storage.azure.com";
+
+            string[] initialMismatchedScopes = new string[] { initialMismatchedScope };
+            string[] serviceChallengeResponseScopes = new string[] { serviceChallengeResponseScope + "/.default" };
+            MockCredential mockCredential = new MockCredential()
+            {
+                GetTokenCallback = (trc, _) =>
+                {
+                    Assert.AreEqual(serviceChallengeResponseScopes, trc.Scopes);
```

Comment:
We should validate that the callback is only called once even if there are multiple requests sent to validate that we are caching properly.

Reply:
fixed.